HierarchicalASTVisitorTest fails

Regression from https://github.com/eclipse-jdt/eclipse.jdt.core/pull/328 

https://download.eclipse.org/eclipse/downloads/drops4/I20220823-1800/testresults/html/org.eclipse.jdt.ui.tests_ep425I-unit-cen64-gtk3-java18_linux.gtk.x86_64_18.html

HierarchicalASTVisitor must be updated to reflect a change in the ASTNode hierarchy. No method visit(AbstractTextElement) was found in HierarchicalASTVisitor.

java.lang.AssertionError: HierarchicalASTVisitor must be updated to reflect a change in the ASTNode hierarchy. No method visit(AbstractTextElement) was found in HierarchicalASTVisitor.
at org.junit.Assert.fail(Assert.java:89)
at org.eclipse.jdt.ui.tests.core.HierarchicalASTVisitorTest.checkHierarchicalASTVisitorMethodExistsFor(HierarchicalASTVisitorTest.java:522)
at org.eclipse.jdt.ui.tests.core.HierarchicalASTVisitorTest.checkAllMethodsForHierarchyExist(HierarchicalASTVisitorTest.java:495)
at org.eclipse.jdt.ui.tests.core.HierarchicalASTVisitorTest.test(HierarchicalASTVisitorTest.java:471)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)

--- a/org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/dom/HierarchicalASTVisitor.java
+++ b/org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/dom/HierarchicalASTVisitor.java
@@ -1185,14 +1185,32 @@ public abstract class HierarchicalASTVisitor extends ASTVisitor {
 		endVisit((AbstractTagElement)node);
 	}
 
+	public boolean visit(AbstractTextElement node) {
+		return visit((ASTNode)node);
+	}
+
+	public void endVisit(AbstractTextElement node) {
+		endVisit((ASTNode)node);
+	}
+
 	@Override
 	public boolean visit(TextElement node) {
-		return visit((ASTNode)node);
+		return visit((AbstractTextElement)node);
 	}
 
 	@Override
 	public void endVisit(TextElement node) {
-		endVisit((ASTNode)node);
+		endVisit((AbstractTextElement)node);
+	}
+
+	@Override
+	public boolean visit(JavaDocTextElement node) {
+		return visit((AbstractTextElement)node);
+	}
+
+	@Override
+	public void endVisit(JavaDocTextElement node) {
+		endVisit((AbstractTextElement)node);
 	}
 
 	@Override

--- a/org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/core/HierarchicalASTVisitorTest.java
+++ b/org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/core/HierarchicalASTVisitorTest.java
@@ -30,6 +30,7 @@ import org.eclipse.core.runtime.Assert;
 import org.eclipse.jdt.core.dom.ASTNode;
 import org.eclipse.jdt.core.dom.ASTVisitor;
 import org.eclipse.jdt.core.dom.AbstractTagElement;
+import org.eclipse.jdt.core.dom.AbstractTextElement;
 import org.eclipse.jdt.core.dom.AbstractTypeDeclaration;
 import org.eclipse.jdt.core.dom.AnnotatableType;
 import org.eclipse.jdt.core.dom.Annotation;
@@ -146,6 +147,24 @@ public class HierarchicalASTVisitorTest {
 		}
 		@SuppressWarnings("unused") // called reflectively
 		public void superEndVisit(AbstractTagElement node) {
+			super.visit(node);
+		}
+
+		@Override
+		public boolean visit(AbstractTextElement node) {
+			registerCall(AbstractTextElement.class);
+			return false;
+		}
+		@SuppressWarnings("unused") // called reflectively
+		public void superVisit(AbstractTextElement node) {
+			super.visit(node);
+		}
+		@Override
+		public void endVisit(AbstractTextElement node) {
+			registerCall(AbstractTextElement.class);
+		}
+		@SuppressWarnings("unused") // called reflectively
+		public void superEndVisit(AbstractTextElement node) {
 			super.visit(node);
 		}
 

